connector.go

```diff
@@ -41,8 +41,10 @@ type Connector struct {
 	EventSink chan<- protoreflect.ProtoMessage
 }
 
+type Option func(*Connector)
+
 // NewConnector returns a base connector implementation that other connectors can embed to add on to.
-func NewConnector() (*Connector, error) {
+func NewConnector(options ...Option) (*Connector, error) {
```

Comment:
Why not just pass in `(m *Manifest)`? It would be much simpler.

Reply:
By using `...Option`, the parameters are optional, so all other connectors that use this function can remain as usual.